features/step_definitions/web_ext_steps.rb

```diff
@@ -13,17 +13,17 @@
 
 Then /^I should see a link to "([^"]*)" with url (for|that starts with|that ends with) "([^"]*)"$/ do |name, attribute_selector, url|
   operator =
-      case attribute_selector
-        when 'that starts with' then '^='
-        when 'that ends with' then '$='
-        else '='
-      end
+    case attribute_selector
+    when 'that starts with' then /^#{Regexp.escape(url)}.*$/
+    when 'that ends with' then /.*#{Regexp.escape(url)}$/
```

Comment:
Because the beginning/end anchors are all that's needed, these can be simplified as:
```ruby
    when 'that starts with' then /^#{Regexp.escape(url)}/
    when 'that ends with' then /#{Regexp.escape(url)}$/
```